src/app/clusters/ota-requestor/OTARequestor.h

```diff
@@ -59,20 +66,35 @@ class OTARequestor : public OTARequestorInterface
     // The BDXDownloader instance should already have the ImageProcessingDelegate set.
     void SetBDXDownloader(chip::BDXDownloader * downloader) { mBdxDownloader = downloader; }
 
-    // Application directs the Requestor to abort any processing related to
-    // the image update
+    // Application directs the Requestor to abort the download in progress. All the Requestor state (such
+    // as the QueryImageResponse content) is preserved
     void AbortImageUpdate();
 
+    // Application directs the Requestor to abort the download in progress. All the Requestor state is
+    // cleared, UploadState is reset to Idle
+    void AbortAndResetState();
+
+    // Application notifies the Requestor on the user consent action, TRUE if consent is given,
+    // FALSE otherwise
+    void OnUserConsent(bool result);
+
+    // Application directs the Requestor to download the image using the suppiled parameter and without
+    // issuing QueryImage
+    OTATriggerResult ResumeImageDownload(const BdxDownloadParameters & bdxParameters)
```

Comment:
You are right, not all of these parameters are needed. I wanted to err on the side of caution and potentially include unneeded parameters rather than omit something that might turn out to be required. 

Having said that, I thought that in addition to the URI at least the Version and the Token should be saved (or at least passed) to help the platform/application keep track of what image is being referenced. 

I can add comments to the rest of the fields saying that we might not need them. In any case, download resumption implementation won't be coming any time soon.